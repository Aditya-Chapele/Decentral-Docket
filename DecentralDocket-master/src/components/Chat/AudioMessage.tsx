/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import React from "react"
import { StyleSheet, TouchableOpacity } from "react-native"
import Svg, { Rect } from "react-native-svg"
import { commonColors } from "../../../docs/config"
import AntDesign from "react-native-vector-icons/AntDesign"
import { heightPercentageToDP as hp } from "react-native-responsive-screen"
import { HStack } from "native-base"
import { IMessage } from "../../stores/chatStore"

//interfaces and types
interface IAudioMessage {
  onLongPress: (message: IMessage) => void
  onPress: () => void
  waveform: Array<number>
  message: IMessage
}
//interfaces and types

//UI Component for messages with audio files
export const AudioMessage: React.FC<IAudioMessage> = ({
  onLongPress,
  onPress,
  waveform,
  message,
}) => {
  return (
    <TouchableOpacity
      accessibilityLabel="Play Audio Message"
      onLongPress={() => onLongPress(message)}
      onPress={() => onPress()}
      activeOpacity={0.7}
      style={styles.button}
    >
      <HStack
        alignItems={"center"}
        style={{
          marginTop: 10,
        }}
      >
        <AntDesign
          name="play"
          size={hp("3%")}
          color={"white"}
          style={{
            marginRight: 4,
            marginLeft: 10,
          }}
        />
        {!!waveform && typeof waveform === "object" && (
          <Svg stroke={commonColors.primaryDarkColor} width={100} height={55}>
            {waveform?.map((item, i) => (
              <Rect
                fill={"rgba(255,255,255,0.9)"}
                key={i}
                width={3}
                x={i * 4}
                y={35 + item}
                height={item === 0 ? -3 : -item * 25}
              />
            ))}
          </Svg>
        )}
      </HStack>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 5,
    height: hp("5%"),
    justifyContent: "center",
    position: "relative",
  },
})
