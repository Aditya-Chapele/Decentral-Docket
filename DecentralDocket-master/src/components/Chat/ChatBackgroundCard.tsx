import { Box, Pressable } from "native-base"
import * as React from "react"
import { heightPercentageToDP as hp } from "react-native-responsive-screen"
import { commonColors } from "../../../docs/config"
import AntDesign from "react-native-vector-icons/AntDesign"
import FastImage from "react-native-fast-image"

interface ChatBackgroundCardProps {
  value?: string
  alt?: string
  isSelected?: boolean
  onSelect: any
  index: number
}

const ChatBackgroundCard = (props: ChatBackgroundCardProps) => {
  return (
    <Pressable
      justifyContent={"center"}
      alignItems={"center"}
      onPress={() => props.onSelect(props.index)}
    >
      <Box
        borderWidth={props.isSelected ? 2 : 0}
        borderColor={props.isSelected ? commonColors.primaryDarkColor : null}
        opacity={props.isSelected ? 20 : 100}
        zIndex={-1}
        h={hp("40%")}
        w={hp("20%")}
        borderRadius={5}
        justifyContent="center"
        alignItems={"center"}
      >
        <FastImage
          style={{
            height: hp("40%"),
            width: hp("20%"),
            borderRadius: 5,
          }}
          source={{ uri: props.value }}
        />
      </Box>
      {props.isSelected ? (
        <AntDesign
          color={"black"}
          name="checkcircle"
          size={hp("3%")}
          style={{ position: "absolute", zIndex: 1 }}
        />
      ) : null}
    </Pressable>
  )
}

export default ChatBackgroundCard
