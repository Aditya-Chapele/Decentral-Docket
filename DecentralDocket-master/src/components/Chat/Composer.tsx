import React, { useMemo } from "react"
import { Composer, ComposerProps } from "react-native-gifted-chat"
import {
  MentionPartType,
  PartType,
  Position,
  Suggestion,
} from "../../helpers/chat/inputTypes"
import {
  generateValueFromPartsAndChangedText,
  generateValueWithAddedSuggestion,
  getMentionPartSuggestionKeywords,
  isMentionPartType,
  parseValue,
} from "../../helpers/chat/inputUtils"

interface IChatComposer extends ComposerProps {
  partTypes: PartType[]
  selection: Position
  onTextChanged: (text: string) => void
  text: string
}

export const ChatComposer: React.FC<IChatComposer> = ({
  partTypes,
  onTextChanged,
  selection,
  text,
  ...props
}) => {
  const { plainText, parts } = useMemo(
    () => parseValue(text, partTypes),
    [text, partTypes]
  )
  const onChangeInput = (changedText: string) => {
    onTextChanged(
      generateValueFromPartsAndChangedText(parts, plainText, changedText)
    )
  }
  const keywordByTrigger = useMemo(() => {
    return getMentionPartSuggestionKeywords(
      parts,
      plainText,
      selection,
      partTypes
    )
  }, [parts, plainText, selection, partTypes])

  const onSuggestionPress =
    (mentionType: MentionPartType) => (suggestion: Suggestion) => {
      const newValue = generateValueWithAddedSuggestion(
        parts,
        mentionType,
        plainText,
        selection,
        suggestion
      )

      if (!newValue) {
        return
      }
      onTextChanged(newValue)
    }
  const renderMentionSuggestions = (mentionType: MentionPartType) => {
    return (
      <React.Fragment key={mentionType.trigger}>
        {mentionType.renderSuggestions &&
          mentionType.renderSuggestions({
            keyword: keywordByTrigger[mentionType.trigger],
            onSuggestionPress: onSuggestionPress(mentionType),
          })}
      </React.Fragment>
    )
  }
  return (
    <>
      {(
        partTypes.filter(
          (one) => isMentionPartType(one) && one.renderSuggestions != null
        ) as MentionPartType[]
      ).map(renderMentionSuggestions)}
      <Composer
        text={plainText}
        multiline={true}
        onTextChanged={(changedText) => onChangeInput(changedText)}
        {...props}
      />
    </>
  )
}
