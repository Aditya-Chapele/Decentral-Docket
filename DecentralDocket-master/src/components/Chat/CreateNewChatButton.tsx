import React from "react"
import { Box, HStack, Text, View } from "native-base"
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from "react-native-responsive-screen"
import AntDesign from "react-native-vector-icons/AntDesign"
import { textStyles } from "../../../docs/config"
import { TouchableOpacity } from "react-native"

export interface ICreateNewChatButton {
  onPress: () => void
}

export const CreateNewChatButton: React.FC<ICreateNewChatButton> = ({
  onPress,
}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        padding: 10,
        paddingLeft: 15,
        minHeight: hp("15%"),
        width: wp("100%"),
      }}
    >
      <HStack alignItems={"center"}>
        <Box
          w={hp("5.5%")}
          h={hp("5.5%")}
          bg={"#64BF7C"}
          rounded="full"
          justifyContent={"center"}
          alignItems="center"
          marginRight={2}
        >
          <AntDesign name="plus" color={"#FFF"} size={hp("4.3%")} />
        </Box>
        <View>
          <Text
            fontSize={hp("2%")}
            fontFamily={textStyles.boldFont}
            _dark={{
              color: "warmGray.50",
            }}
            color="coolGray.800"
          >
            Create a new room
          </Text>
          <Text
            fontFamily={textStyles.regularFont}
            fontSize={hp("1.5%")}
            color="coolGray.600"
            _dark={{
              color: "warmGray.100",
            }}
          >
            Your own room, share with anyone you like
          </Text>
        </View>
      </HStack>
    </TouchableOpacity>
  )
}
