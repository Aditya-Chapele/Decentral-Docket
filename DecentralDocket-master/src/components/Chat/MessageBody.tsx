/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import PropTypes from "prop-types"
import React from "react"
import { View, ViewPropTypes, StyleSheet } from "react-native"

import { Avatar, Day, utils, SystemMessage } from "react-native-gifted-chat"
import { textStyles } from "../../../docs/config"
import Bubble from "./MessageBubble"

const { isSameUser, isSameDay } = utils
export default class Message extends React.Component {
  shouldComponentUpdate(nextProps) {
    const next = nextProps.currentMessage
    const current = this.props.currentMessage

    const nextPropsPreviousMessage = nextProps.previousMessage
    if (next.tokenAmount !== current.tokenAmount) return true

    if (next.numberOfReplies !== current.numberOfReplies) return true

    if (next.text !== current.text) return true

    return false
  }
  getInnerComponentProps() {
    const { containerStyle, ...props } = this.props
    return {
      ...props,
      isSameUser,
      isSameDay,
      containerStyle,
    }
  }

  renderDay() {
    if (this.props.currentMessage.createdAt) {
      const dayProps = this.getInnerComponentProps()
      if (this.props.renderDay) {
        return this.props.renderDay(dayProps)
      }
      return <Day {...dayProps} />
    }
    return null
  }

  renderBubble() {
    const { containerStyle, ...props } = this.props
    if (this.props.renderBubble) {
      return this.props.renderBubble(props)
    }
    // @ts-ignore
    return (
      <Bubble
        usernameStyle={{ fontFamily: textStyles.regularFont }}
        {...props}
      />
    )
  }

  renderAvatar() {
    const { containerStyle, ...props } = this.props
    return (
      <View accessibilityLabel="User photo (tap to view profile)">
        <Avatar {...props} />
      </View>
    )
  }

  renderSystemMessage() {
    const { containerStyle, ...props } = this.props
    if (this.props.renderSystemMessage) {
      return this.props.renderSystemMessage(props)
    }
    return <SystemMessage {...props} textStyle={{ textAlign: "center" }} />
  }

  render() {
    const { currentMessage, nextMessage, position, containerStyle } = this.props
    if (currentMessage) {
      const sameUser = isSameUser(currentMessage, nextMessage)
      return (
        <View accessibilityLabel="Message (long tap to interact)">
          {this.renderDay()}
          {currentMessage.system ? (
            this.renderSystemMessage()
          ) : (
            <View
              style={[
                styles[position].container,
                { marginBottom: sameUser ? 2 : 10 },
                !this.props.inverted && { marginBottom: 2 },
                containerStyle && containerStyle[position],
              ]}
            >
              {this.props.position === "left" ? this.renderAvatar() : null}
              {this.renderBubble()}
              {this.props.position === "right" ? this.renderAvatar() : null}
            </View>
          )}
        </View>
      )
    }
    return null
  }
}

const styles = {
  left: StyleSheet.create({
    container: {
      flexDirection: "row",
      alignItems: "flex-end",
      justifyContent: "flex-start",
      marginLeft: 8,
      marginRight: 0,
      fontFamily: textStyles.regularFont,
    },
  }),
  right: StyleSheet.create({
    container: {
      flexDirection: "row",
      alignItems: "flex-end",
      justifyContent: "flex-end",
      marginLeft: 0,
      marginRight: 8,
      fontFamily: textStyles.regularFont,
    },
  }),
}

Message.defaultProps = {
  renderAvatar: undefined,
  renderBubble: null,
  renderDay: null,
  renderSystemMessage: null,
  position: "left",
  currentMessage: {},
  nextMessage: {},
  previousMessage: {},
  user: {},
  containerStyle: {},
  showUserAvatar: false,
  inverted: true,
  shouldUpdateMessage: undefined,
}

Message.propTypes = {
  renderAvatar: PropTypes.func,
  renderBubble: PropTypes.func,
  renderDay: PropTypes.func,
  currentMessage: PropTypes.object,
  nextMessage: PropTypes.object,
  previousMessage: PropTypes.object,
  user: PropTypes.object,
  containerStyle: PropTypes.shape({
    left: ViewPropTypes.style,
    right: ViewPropTypes.style,
  }),
}
