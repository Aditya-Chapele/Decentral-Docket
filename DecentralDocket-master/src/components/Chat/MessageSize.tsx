/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import { Box, Text } from "native-base"
import React from "react"
import { heightPercentageToDP as hp } from "react-native-responsive-screen"
import Ionicons from "react-native-vector-icons/Ionicons"

import { StyleSheet } from "react-native"
export const MessageSize = ({
  size,
  duration,
}: {
  size: string
  duration?: string
}) => {
  return (
    <Box>
      <Box
        justifyContent={"center"}
        alignItems={"center"}
        flexDirection={"row"}
        bg={"rgba(0,0,0,0.6)"}
        rounded={"md"}
        style={styles.container}
      >
        <Ionicons name="arrow-down-outline" size={hp("1.7%")} color={"white"} />
        <Text color={"white"} fontSize={hp("1.6%")}>
          {size}
        </Text>
      </Box>

      {duration && (
        <Text fontSize={hp("1.6%")} color={"white"}>
          {duration}
        </Text>
      )}
    </Box>
  )
}

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    top: 10,
    left: 10,
    zIndex: 999,
  },
})
