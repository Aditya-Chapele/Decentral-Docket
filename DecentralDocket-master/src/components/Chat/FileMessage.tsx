/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import { Box } from "native-base"
import React from "react"
import FastImage from "react-native-fast-image"
import { TouchableOpacity } from "react-native-gesture-handler"
import { heightPercentageToDP as hp } from "react-native-responsive-screen"
import { formatBytes } from "../../helpers/chat/formatBytes"
import { MessageSize } from "./MessageSize"

interface FileMessageProps {
  url: string
  size: string
  onLongPress?: () => void
  onPress: () => void
}

export const FileMessage = ({
  size,
  onLongPress,
  onPress,
}: FileMessageProps) => {
  const formatedSize = formatBytes(parseFloat(size), 2)
  return (
    <TouchableOpacity
      onLongPress={onLongPress}
      onPress={onPress}
      activeOpacity={0.7}
      style={{
        borderRadius: 5,
        justifyContent: "center",
        position: "relative",
      }}
    >
      {size && (
        <MessageSize size={formatedSize.size + " " + formatedSize.unit} />
      )}
      <Box p={"1.5"}>
        <FastImage
          style={{
            width: hp("22%"),
            height: hp("22%"),
            borderRadius: 10,
            borderColor: "white",
          }}
          source={require("../../assets/fileIcon.png")}
        />
      </Box>
    </TouchableOpacity>
  )
}
