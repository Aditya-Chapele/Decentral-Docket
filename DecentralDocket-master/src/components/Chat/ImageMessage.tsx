/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import { Box, Text } from "native-base"
import React from "react"
import FastImage from "react-native-fast-image"
import { TouchableOpacity } from "react-native-gesture-handler"
import { heightPercentageToDP as hp } from "react-native-responsive-screen"
import { textStyles } from "../../../docs/config"
import { formatBytes } from "../../helpers/chat/formatBytes"
import { MessageSize } from "./MessageSize"

//interface and types
interface ImageMessageProps {
  url: any
  size: any
  onLongPress?: () => void
  onPress: any
  nftId?: string
  nftName?: string
}
//interface and types

//component to render UI for messages that have an image in it.
export const ImageMessage = ({
  url,
  size,
  onLongPress,
  onPress,
  nftId,
  nftName,
}: ImageMessageProps) => {
  const formatedSize = formatBytes(parseFloat(size), 2)
  return (
    <>
      {!!nftId && (
        <Text
          color={"white"}
          fontWeight={"bold"}
          fontFamily={textStyles.boldFont}
        >
          🔗🖼️ NFT: {nftName}
        </Text>
      )}
      <TouchableOpacity
        onLongPress={onLongPress}
        accessibilityLabel={"Image Attachment"}
        onPress={onPress}
        activeOpacity={0.7}
        style={{
          borderRadius: 5,
          justifyContent: "center",
          position: "relative",
        }}
      >
        {size && (
          <MessageSize size={formatedSize.size + " " + formatedSize.unit} />
        )}
        {!!nftId && <MessageSize size={"NFT"} />}
        <Box p={"1.5"}>
          <FastImage
            style={{
              width: hp("22%"),
              height: hp("22%"),
              borderRadius: 10,
              borderColor: "white",
              borderWidth: nftId ? 2 : 0,
            }}
            source={{ uri: url }}
          />
        </Box>
      </TouchableOpacity>
    </>
  )
}
