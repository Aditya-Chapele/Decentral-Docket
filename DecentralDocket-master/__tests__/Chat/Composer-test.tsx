import React from "react"
import renderer from "react-test-renderer"

import { ChatComposer } from "../../src/components/Chat/Composer"
import { PartType } from "../../src/helpers/chat/inputTypes"

const mockPartTypes: PartType[] = [
  {
    trigger: "@", // Should be a single character like '@' or '#'
  },
]

test("renders correctly", () => {
  const tree = renderer
    .create(
      <ChatComposer
        onTextChanged={(text) => console.log(text)}
        partTypes={mockPartTypes}
        selection={{ start: 0, end: 0 }}
        text="dsds"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
