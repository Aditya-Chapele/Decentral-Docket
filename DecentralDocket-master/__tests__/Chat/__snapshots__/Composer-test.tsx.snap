// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  accessibilityLabel="Type a message..."
  accessible={true}
  autoFocus={false}
  editable={true}
  enablesReturnKeyAutomatically={true}
  keyboardAppearance="default"
  multiline={true}
  onChange={[Function]}
  onChangeText={[Function]}
  onContentSizeChange={[Function]}
  placeholder="Type a message..."
  placeholderTextColor="#b2b2b2"
  style={
    [
      {
        "flex": 1,
        "fontSize": 16,
        "lineHeight": 16,
        "marginBottom": 5,
        "marginLeft": 10,
        "marginTop": 6,
      },
      {},
      {
        "height": 33,
      },
    ]
  }
  testID="Type a message..."
  underlineColorAndroid="transparent"
  value="dsds"
/>
`;
